shared/src/ledger/eth_bridge/vp/mod.rs

```diff
@@ -27,6 +28,56 @@ where
     pub ctx: Ctx<'ctx, DB, H, CA>,
 }
 
+impl<'ctx, DB, H, CA> EthBridge<'ctx, DB, H, CA>
+where
+    DB: 'static + ledger_storage::DB + for<'iter> ledger_storage::DBIter<'iter>,
+    H: 'static + StorageHasher,
+    CA: 'static + WasmCacheAccess,
+{
+    /// If the bridge's escrow key was changed, we check
+    /// that the balance increased and that the bridge pool
+    /// VP has been triggered. The bridge pool VP will carry
+    /// out the rest of the checks.
+    fn check_escrow(&self, verifiers: &BTreeSet<Address>) -> bool {
+        let escrow_key = balance_key(&xan(), &super::ADDRESS);
+        let escrow_pre: Amount = if let Ok(Some(bytes)) =
+            self.ctx.read_pre(&escrow_key)
+        {
+            BorshDeserialize::try_from_slice(bytes.as_slice())
+                .expect("Deserializing a balance from storage shouldn't fail")
+        } else {
+            tracing::debug!(
+                "Could not retrieve the Ethereum bridge VP's balance from \
+                 storage"
+            );
+            return false;
+        };
+        let escrow_post: Amount = if let Ok(Some(bytes)) =
+            self.ctx.read_pre(&escrow_key)
+        {
+            BorshDeserialize::try_from_slice(bytes.as_slice())
+                .expect("Deserializing a balance from storage shouldn't fail")
+        } else {
+            tracing::debug!(
+                "Could not retrieve the modified Ethereum bridge VP's balance \
+                 after applying tx"
+            );
+            return false;
+        };
+
+        // The amount escrowed should increase.
+        if escrow_pre < escrow_post {
+            verifiers.contains(&storage::bridge_pool::BRIDGE_POOL_ADDRESS)
+        } else {
+            tracing::info!(
+                "A normal tx cannot decrease the amount of Nam escrowed in \
+                 the Ethereum bridge"
+            );
+            false
```

Comment:
In theory some one could overwrite the same value and it would then show up in pre and post reads. Escrowing 0 tokens serves no purpose, so I decided to not allow it.